lib/sshutils/ctx.go

```diff
@@ -85,8 +85,23 @@ type agentChannel struct {
 	ch ssh.Channel
 }
 
+// Close closes the agent channel.
 func (a *agentChannel) Close() error {
-	return a.ch.Close()
+	// For graceful teardown, close the write part of the channel first. This
+	// will send "EOF" packet (type 96) to the other side which will drain and
+	// close the channel.
+	//
+	// The regular close after that will send "close" packet (type 97) which
+	// won't attempt to send us any more data since the channel is already
+	// closed.
```

Comment:
Could we add a test for this?

Reply:
I've been trying to think of a nice way to test this properly but have not been able to come up with a good way to do this. Do you have any ideas how I could go about doing this?

Reply:
I've added a test that will at least break if someone swaps these lines around,  still trying to figure out a nice way to mock everything required if we want to test the packets sent in the crypto/ssh package are what we expect

Reply:
My original idea was that you could open an actual SSH connection and confirm the last 2 bytes sent, but your test at least confirms that Teleport closes things in the right order, so I think it's good enough as long as you've also confirmed the fix manually